packages/adapter/addon/-private/utils/serialize-query-params.ts

```diff
@@ -41,7 +41,7 @@ export function serializeQueryParams(queryParamsObject: object | string): string
     return s;
   }
 
-  return buildParams('', queryParamsObject).join('&').replace(/%20/g, '+');
```

Comment:
re: https://github.com/emberjs/data/issues/7824#issuecomment-1003642900

@alechirsch I noticed this is only invoked with a GET or HEAD request.  So according to the [spec](https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1), the `+` sign should only be encoded with a POST `application/x-www-form-urlencoded` request.  Which would not be this case.  

This is in fact a bug and we should be slightly worried that this might cause issues for users though.

https://github.com/emberjs/data/blob/e076e0ae71ae6426ca53ad3c5501a0af7ceca883/packages/adapter/addon/rest.ts#L1499